diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/PackageNaming.kt

```diff
@@ -209,50 +192,38 @@ class PackageNaming(private val configRules: List<RulesConfig>) : Rule("package-
+                ?: packageDirectiveNode.findChildByType(REFERENCE_EXPRESSION)
 
-            // putting composite node first in the parent tree and adding IDENTIFIER and DOT as children to it after
-            parentNode.addChild(compositeElementWithNameAndDot!!, insertBeforeNode)
-            compositeElementWithNameAndDot!!.addChild(childPackageNamePart!!)
-            compositeElementWithNameAndDot!!.addChild(childDot!!)
-        }
+        val generatedPackageDirective = KotlinParser()
+                .createNode("$PACKAGE_KEYWORD $packageName", true)
 
-        // removing extra DOT that is not needed if we were inserting the whole package name (not just a part) to the parent
-        if (insertBeforeNode == null) {
-            compositeElementWithNameAndDot!!.removeChild(childDot!!)
+        if (packageNameNode == null) {
+            // there is missing package statement in a file, so it will be created and inserted
+            val newPackageDirective = generatedPackageDirective.findLeafWithSpecificType(PACKAGE_DIRECTIVE)!!
+            packageDirectiveNode.treeParent.replaceChild(packageDirectiveNode, newPackageDirective)
+            newPackageDirective.treeParent.addChild(PsiWhiteSpaceImpl("\n\n"), newPackageDirective.treeNext)
+
+        } else {
+            // simply replacing only node connected with the package name, all other nodes remain unchanged
+            packageDirectiveNode.replaceChild(packageNameNode, generatedPackageDirective.findLeafWithSpecificType(DOT_QUALIFIED_EXPRESSION)!!)
         }
+
     }
 
     /**
      * checking and fixing package directive if it does not match with the directory where the file is stored
      */
-    private fun checkFilePathMatchesWithPackageName(packageNameParts: List<ASTNode>, realName: List<String>) {
-        if (realName.isNotEmpty() && packageNameParts.map { node -> node.text } != realName) {
-            PACKAGE_NAME_INCORRECT_PATH.warnAndFix(configRules, emitWarn, isFixMode, realName.joinToString(PACKAGE_SEPARATOR),
-                    packageNameParts[0].startOffset, packageNameParts[0]) {
-                // need to get first top-level DOT-QUALIFIED-EXPRESSION
-                // -- PACKAGE_DIRECTIVE
-                //    -- DOT_QUALIFIED_EXPRESSION
-                val parentNode = packageNameParts[0]
-                    .parent(PACKAGE_DIRECTIVE)!!
-                    .findChildByType(DOT_QUALIFIED_EXPRESSION)!!
+    private fun checkFilePathMatchesWithPackageName(packageNameParts: List<ASTNode>, realName: List<String>, packageDirective: ASTNode) {
 
-                parentNode.getChildren(null).forEach { node -> parentNode.removeChild(node) }
-                createAndInsertPackageName(parentNode, null, realName)
+        val realPackageNameStr = realName.joinToString(PACKAGE_SEPARATOR)
+        val offset = packageNameParts[0].startOffset
+        if (realName.isNotEmpty() && packageNameParts.map { node -> node.text } != realName) {
```

Comment:
```suggestion
        if (realName.isNotEmpty() && packageNameParts.map { node -> node.text } != realName) {
            val realPackageNameStr = realName.joinToString(PACKAGE_SEPARATOR)
            val offset = packageNameParts[0].startOffset
```
to be more consistent with our code style recommendations on local variables declaration

Reply:
that's true, that should be checked by diktat 👍